src/strategies/FluentStrategy.ts

```diff
@@ -0,0 +1,104 @@
+import {Arbitrary, FluentPick} from '../arbitraries'
+import {FluentConfig, FluentResult} from '../FluentCheck'
+import {StrategyArbitraries} from './types'
+
+export interface FluentStrategyInterface {
+  hasInput: <K extends string>(arbitraryName: K) => boolean
+  getInput: <K extends string, A>(arbitraryName: K) => FluentPick<A>
+  handleResult: () => void
+}
+
+export class FluentStrategy implements FluentStrategyInterface {
+
+  /**
+   * Record of all the arbitraries used for composing a given test case.
+   */
+  public arbitraries: StrategyArbitraries = {}
+
+  /**
+   * Default constructor. Receives the FluentCheck configuration, which is used for test case generation purposes.
+   */
+  constructor(public readonly configuration: FluentConfig = {sampleSize: 1000, shrinkSize: 500}) {}
+
+  /**
+   * Allows the configuration of a given strategy
+   */
+  config(config: FluentConfig) {
+    for (const k in config) this.configuration[k] = config[k]
+    return this
+  }
```

Comment:
Maybe:

```typescript
config(config: Partial<FluentConfig>) {
    this.configuration = {...this.configuration, ...config}
    return this
}
```

So that config can be called with just one parameter, for example.

Reply:
That could be a nice implementation even for future types of configurations. I will change that!